usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -2931,9 +2943,34 @@ arc_evict_hdr(arc_buf_hdr_t *hdr, kmutex_t *hash_lock)
 	return (bytes_evicted);
 }
 
+/*
+ * Based on l2arc_spa_list, returns true if the
+ * given spa has an alive (!dead) L2 device,
+ * false otherwise.
+ */
+static boolean_t
+l2arc_alive(uint64_t spa)
```

Comment:
I could, but for each buffer to evict, we would have to :
- browse the list of vdev ;
- and for each one of these vdevs, test if its spa matches the expected one, and test if it is non-faulted (`vdev_is_dead`).

I think this could be pretty time consuming, this is why I preferred to make `l2arc_dev_get_next()` maintaining a list of spa with alive L2 vdevs, as this function already make all the tests we need, on a regular basis. We then just have to walk the list of spa with alive L2 vdev (`l2arc_spa_list`) for each buffer to evict.